junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -122,13 +123,17 @@ private TestTemplateInvocationContext createInvocationContext(ParameterizedTestN
 		return new ParameterizedTestInvocationContext(formatter, methodContext, arguments);
 	}
 
-	private ParameterizedTestNameFormatter createNameFormatter(Method templateMethod,
+	private ParameterizedTestNameFormatter createNameFormatter(ExtensionContext extensionContext, Method templateMethod,
 			ParameterizedTestMethodContext methodContext, String displayName, int argumentMaxLength) {
 		ParameterizedTest parameterizedTest = findAnnotation(templateMethod, ParameterizedTest.class).get();
-		String pattern = Preconditions.notBlank(parameterizedTest.name().trim(),
-			() -> String.format(
-				"Configuration error: @ParameterizedTest on method [%s] must be declared with a non-empty name.",
-				templateMethod));
+		// the priority of pattern is shown below.
+		// 1) configuration parameter
+		// 2) ParameterizedTest name
+		String pattern = extensionContext.getConfigurationParameter(DISPLAY_NAME_TEMPLATE_KEY).map(String::trim).orElse(
+			Preconditions.notBlank(parameterizedTest.name().trim(),
+				() -> String.format(
+					"Configuration error: @ParameterizedTest on method [%s] must be declared with a non-empty name.",
+					templateMethod)));
```

Comment:
Please only call `trim()` once and add validation (with an appropriate message) for the config parameter along with a test case each for both scenarios.

Reply:
As mentioned in https://github.com/junit-team/junit5/pull/2532/files#r559023221 and subsequent comments, the priority needs to be reversed for backwards compatibility and local overrides of the new global default.